ml-algorithms/src/test/java/org/opensearch/ml/engine/utils/ZipUtilsTest.java

```diff
@@ -0,0 +1,39 @@
+package org.opensearch.ml.engine.utils;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+public class ZipUtilsTest {
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Test
+    public void testEmptyZipFile() throws IOException {
+        exceptionRule.expect(RuntimeException.class);
+        Path path = Paths.get("build/empty.zip");
+        File file = new File(path.toUri());
+        Path output = Paths.get("build/output");
+        Files.createDirectories(output);
+        ZipUtils.unzip(file, output);
+    }
+
+    @Test
+    public void testUnzipFile() throws IOException, URISyntaxException {
+        File testZipFile = new File(Objects.requireNonNull(getClass().getResource("foo.zip")).toURI());
```

Comment:
We already have multiple zip files in the same resource folder. Do we need to add another zip file only for this test?

Reply:
I think the answer is yes because we cannot ensure these zip file will be there forever with same version or name.

Reply:
For example, `all-MiniLM-L6-v2_onnx.zip` is being used in 3 difference tests. So I don't think this file will be disappear suddenly. Even if somebody replace this file, that will fail the test and they can update the changed file name accordingly.

In addition, in production this util will unzip a model zip file then why not use the similar file in test...

Reply:
Meanwhile I want to build a small size folder that has an empty sub-folder, a non-empty sub-folder and a file to cover most cases, I don't think we have that kind of zip file.